themes/default/content/docs/languages-sdks/javascript/_index.md

```diff
@@ -183,13 +192,26 @@ runtime:
 
 ## Package Management
 
-Pulumi has official support for NPM and Yarn Classic. Pulumi does
-not support Yarn Plug'n'Play.
+Pulumi has official support for NPM, PNPM, and Yarn Classic. NPM is the default package manager.
+Pulumi does not support Yarn Plug'n'Play.
+
+Users can force Pulumi to use a specific package manager via environment variable.
+Pulumi will use Yarn Classic if `PULUMI_PREFER_YARN=true`, and PNPM if
+`PULUMI_PREFER_PNPM=true`. Additionally, Pulumi attempts to detect your package manager prefers by looking for
+lockfile in your program directory.
+
+The precedence order is as follows:
+
+1. Setting `PULUMI_PREFER_YARN` or `PULUMI_PREFER_PNPM` has the highest precedence.
+
+2. If Pulumi finds a `yarn.lock` file in the program directory, Yarn Classic will be used.
+
+3. If Pulumi finds `pnpm-lock.yaml` file in the program directory, PNPM will be used.
+
+4. Pulumi will fall back to NPM in all other cases.
 
-Pulumi defaults to using NPM. However, if Pulumi detects a `yarn.lock` file
-in the project root, or the environment variable `PULUMI_PREFER_YARN=true`,
-then Pulumi will use Yarn instead of NPM if the executable is available in the
-path.
+Note that if the hinted package manager is not found in the user's PATH, then Pulumi continues
+through precedence order until a valid package manager is found.
```

Comment:
```suggestion
If the hinted package manager is not found in the user's PATH, then Pulumi continues
in precedence order until a valid package manager is found.
```

or put it into a note shortcode

Reply:
I went with the note shortcode and copy changes.